elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchConnectionSettings.scala

```diff
@@ -17,21 +22,40 @@ final class ElasticsearchConnectionSettings private (
 
   def hasCredentialsDefined: Boolean = username.isDefined && password.isDefined
 
+  def withHeaders(headers: List[HttpHeader]): ElasticsearchConnectionSettings =
+    copy(headers = Option(headers))
+
+  def hasHeadersDefined: Boolean = headers.isDefined
+
+  def withConnectionContext(connectionContext: HttpsConnectionContext): ElasticsearchConnectionSettings =
+    copy(connectionContext = Option(connectionContext))
+
+  def hasConnectionContextDefined: Boolean = connectionContext.isDefined
+
   def copy(baseUrl: String = baseUrl,
```

Comment:
@seglo - should this method be private? Because it is not, this class might end up being restricted for future updates because changes could affect binary compatibility. Note that other classes in this directory have the `copy` method set with private scope. Specifically, the `ElasticsearchSourceSettings` and the `ElasticsearchWriteSettings`. In my understanding, these classes are set up with very intentional member scopeto allow for binary compatible expansions on the alpakka api.

(Sorry, I know this line was not changed in this PR, but I am pretty sure this public `copy` method is the reason that @jabbaugh needed to update the mima filters.)

Reply:
For reference, this class is new to the 3.0.0 release, so I don't think changing this method to `private` qualifies as a break change yet.

Reply:
Yes, I think it's fine to make it private.

Reply:
We'll still need to add the MiMa exceptions since there's been a milestone release, but that's fine too.